mono/metadata/loader.c

```diff
@@ -2106,6 +2121,14 @@ mono_loader_lock_is_owned_by_self (void)
 	return GPOINTER_TO_UINT (TlsGetValue (loader_lock_nest_id)) > 0;
 }
 
+gboolean
+mono_loader_lock_self_is_waiting (void)
+{
+	g_assert (loader_lock_track_ownership);
```

Comment:
I think this is fine, this is the same assert as in mono_loader_lock_is_owned_by_self and both these functions are only used from the debugger agent which calls mono_loader_lock_track_ownership(TRUE) on init.